### PR TITLE
bigloo 4.4b

### PR DESCRIPTION
**Pull request by gromgit**
Created with `brew bump-formula-pr`.

Also:
* added license
* added `openjdk` dependency, since Bigloo is being configured with `--jvm=yes`
* use a bunch of Homebrew libraries instead of depending on vendored ones (which in at least the case of `gmp` causes a install conflict)
* `pkg-config` seems to be a required build dependency

### CODE REVIEW

File: Formula/bigloo.rb [hunk 1]
```diff
@@ -1,10 +1,9 @@
 class Bigloo < Formula
   desc "Scheme implementation with object system, C, and Java interfaces"
   homepage "https://www-sop.inria.fr/indes/fp/Bigloo/"
-  url "ftp://ftp-sop.inria.fr/indes/fp/Bigloo/bigloo4.3e.tar.gz"
-  version "4.3e"
-  sha256 "43363cb968c57925f402117ff8ec4b47189e2747b02350805a34fa617d9f618a"
-  revision 1
+  url "ftp://ftp-sop.inria.fr/indes/fp/Bigloo/bigloo-4.4b.tar.gz"
+  sha256 "a313922702969b0a3b3d803099ea05aca698758be6bd0aae597caeb6895ce3cf"
+  license "GPL-2.0-or-later"
```
Comment (carlocab) [inline Formula/bigloo.rb:6]: How'd you figure the `or-later` part? I looked in the tarball and thought it was `-only`, but maybe I missed something in the docs.

**Comment by carlocab** [comment]
It looks like it built its own `gmp`, hence the link failure.

**Comment by carlocab** [comment]
Try this:
```diff
diff --git a/Formula/bigloo.rb b/Formula/bigloo.rb
index 5ab7c1415bc..ee2c846dc77 100644
--- a/Formula/bigloo.rb
+++ b/Formula/bigloo.rb
@@ -24,6 +24,7 @@ class Bigloo < Formula
 
   depends_on "gmp"
   depends_on "libunistring"
+  depends_on "libuv"
   depends_on "openjdk"
   depends_on "openssl@1.1"
   depends_on "pcre"
@@ -43,6 +44,7 @@ class Bigloo < Formula
       --disable-flac
       --disable-srfi27
       --jvm=yes
+      --custom-all-libs
     ]
 
     system "./configure", *args
```

You can also pass `--custompcre=yes`, `--customgmp=yes`, and `--customlibuv=yes` if you prefer.

**Comment by gromgit** [comment]
> This also may be informative: https://gitlab.inria.fr/mserrano/hopbrew/-/blob/master/Formula/bigloo-latest.rb

I was actually looking at that to rationalize the failures from the 4.4a PR. It looks like the key problem is that the current Homebrew formula uses `--os-macosx`, which actually _forces_ the use of all the vendored libraries.

**Comment by carlocab** [comment]
I think you still want to pass specific `--custom*` flags, though. The `configure` script has most of them set to `no` by default. (e.g. `gmp`, `unistring`, `pcre`) Unless `custom*=yes` here actually means use their vendored library?

**Comment by gromgit** [comment]
> Unless `custom*=yes` here actually means use their vendored library?

Yup.

**Comment by carlocab** [comment]
Right, then I think you'll want `--customlibuv=no`, as that one's `yes` by default.

**Comment by gromgit** [comment]
Hmmm, Bigloo's build system is rather unusual. I should probably get some sleep first before continuing to puzzle this out. :laughing:

File: Formula/bigloo.rb [hunk 2]
```diff
@@ -21,19 +20,30 @@ class Bigloo < Formula
...
   depends_on "gmp"
+  depends_on "libunistring"
+  depends_on "libuv"
+  depends_on "openjdk"
   depends_on "openssl@1.1"
+  depends_on "pcre"
 
   def install
+    inreplace "configure" do |s|
+      s.gsub!(/^\s+custom\w+=(yes)$/, "no")
```
Comment (Bo98) [inline Formula/bigloo.rb:35]:
```suggestion
      s.gsub!(/^(\s+custom\w+)=yes$/, "\\1=no")
```

Comment (Bo98) [reply Formula/bigloo.rb:35]: Can be simplified to one line too problably: `inreplace "configure", /^(\s+custom\w+)=yes$/, "\\1=no"`

**Comment by Bo98** [comment]
Looks like a `configure` bug. I think this:

```sh
$echo "EXTRALIBS+=-l$(GCLIB)" >> $makefile_cfg
```

should be:

```sh
$echo 'EXTRALIBS+=-l$(GCLIB)' >> $makefile_cfg
```

**Comment by carlocab** [comment]
> Looks like a `configure` bug.

Yep, I had surmised as much but didn't have the patience to track it down just yet. Thanks for looking at it.

**Comment by carlocab** [comment]
Nope, no ARM support.

**Review by carlocab — APPROVED** [review]

**Comment by gromgit** [comment]
Thanks for taking up the mantle, @carlocab. Was struggling to stay awake, with builds running on almost 10-year-old hardware. :laughing:

**Comment by carlocab** [comment]
No problem. The build system was intriguing. Plus I've seen `bigloo` too many times when running `brew livecheck`, so it was nice to get this updated.

**Pull request closed without merging**